datastore/bolt/api_key_test.go

```diff
@@ -0,0 +1,303 @@
+//go:build integration
+// +build integration
+
+package bolt
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"testing"
+
+	"github.com/frain-dev/convoy/auth"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/util"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/pbkdf2"
+)
+
+func Test_CreateAPIKey(t *testing.T) {
+	db, closeFn := getDB(t)
+	defer closeFn()
+
+	apiKeyRepo := NewApiRoleRepo(db)
+
+	maskID, salt, encodedKey, err := generateAPIKey()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	role := auth.Role{Type: "super_user"}
+
+	newApiKey := &datastore.APIKey{
+		UID:    uuid.New().String(),
+		MaskID: maskID,
+		Name:   "Api Key",
+		Type:   "test_api_key",
+		Role:   role,
+		Hash:   encodedKey,
+		Salt:   salt,
+	}
+
+	require.NoError(t, apiKeyRepo.CreateAPIKey(context.Background(), newApiKey))
+	apiKey, err := apiKeyRepo.FindAPIKeyByID(context.Background(), newApiKey.UID)
+
+	require.NoError(t, err)
+	require.Equal(t, newApiKey.UID, apiKey.UID)
+
+}
+
+func Test_UpdateAPIKey(t *testing.T) {
+	db, closeFn := getDB(t)
+	defer closeFn()
+
+	apiKeyRepo := NewApiRoleRepo(db)
+
+	maskID, salt, encodedKey, err := generateAPIKey()
+
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Why not use `require.NoError()`
```suggestion
		require.NoError(t, err)
```

Reply:
I don't think that assertion suits this test case, since it's something a bit different.

Reply:
Hmm, Ok